### PR TITLE
Fix CodeQL Integer Conversion Warnings

### PR DESCRIPTION
**Pull request by jentfoo**
These warnings are fairly pedantic in most cases, but it was easiest to just fix them as I reviewed them. I added to `api/utils/conv,go` the boundary checks for doing this casting.  In many cases this allowed a drop in replacement of one of the newly added functions.

Example CodeQL findings fixed by this:
https://github.com/gravitational/teleport/security/code-scanning/81
https://github.com/gravitational/teleport/security/code-scanning/80
https://github.com/gravitational/teleport/security/code-scanning/79
https://github.com/gravitational/teleport/security/code-scanning/78
https://github.com/gravitational/teleport/security/code-scanning/77
https://github.com/gravitational/teleport/security/code-scanning/70
https://github.com/gravitational/teleport/security/code-scanning/69

### CODE REVIEW

File: lib/web/desktop.go [hunk 1]
```diff
@@ -118,11 +118,11 @@ func (h *Handler) createDesktopConnection(
 	if username == "" {
 		return sendTDPError(ws, trace.BadParameter("missing username"))
 	}
-	width, err := strconv.Atoi(q.Get("width"))
+	width, err := apiutils.StrToUInt32(q.Get("width"))
```
Comment (zmb3) [inline lib/web/desktop.go:121]: I don't think this is necessary here. There's already a check for max width height on line 130 (which is a stricter check than max uint32 anyway).

File: api/utils/conv.go [hunk 2]
```diff
@@ -20,10 +20,88 @@ package utils
...
 	"bytes"
 	"encoding/json"
+	"math"
+	"strconv"
 
 	"github.com/gravitational/trace"
 )
 
+// IntToUInt16 converts an int to uint16, checking for possible overflow or underflow issues.
+func IntToUInt16(i int) (uint16, error) {
```
Comment (zmb3) [inline api/utils/conv.go:30]: I'm not a big fan of general purpose functions becoming part of Teleport's public API module. If this is necessary (and I'm not totally convinced of that yet, either), it seems like more of an implementation detail than functions we'd want to expose to people building things with Teleport.

tl;dr, it's hard to remove things from a public API once they exist, so best to keep the surface area as small as reasonably possible.

File: lib/srv/ctx.go [hunk 3]
```diff
@@ -695,7 +694,7 @@ func (c *ServerContext) OpenXServerListener(x11Req x11.ForwardRequestPayload, di
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	originPortI, err := strconv.Atoi(originPort)
+	originPortI, err := apiutils.StrToUInt16(originPort)
```
Comment (zmb3) [inline lib/srv/ctx.go:697]: Similarly here, there's no way this check ever triggers an error since the the host port is coming from an existing conn (meaning the port must be valid). All we've done is added more branches and indirection and not increased security in any meaningful way.

File: lib/srv/reexec.go [hunk 4]
```diff
@@ -875,11 +876,11 @@ func newParker(ctx context.Context, credential syscall.Credential) error {
 // getCmdCredentials parses the uid, gid, and groups of the
 // given user into a credential object for a command to use.
 func getCmdCredential(localUser *user.User) (*syscall.Credential, error) {
-	uid, err := strconv.Atoi(localUser.Uid)
+	uid, err := apiutils.StrToUInt32(localUser.Uid)
```
Comment (zmb3) [inline lib/srv/reexec.go:879]: Is there any possible way that Go's `os/user` package would return a UID that's not valid? I would be surprised if that's the case, meaning this check is also not necessary.

**Review by zmb3 — COMMENTED** [review]
Overall, it feels like we're introducing changes to appease an automated scanner as opposed to changes that demonstrably improve security.

At the end of the day, the source code is meant to be read by humans, not machines, so I can't say I'm in favor introducing more indirection and extra function calls just to make a machine happy at the expense of readability for engineers.

**Comment by jentfoo** [comment]
@zmb3 Sounds good to me.  I agree the changes here are pedantic and not a security risk.  I only wanted to present them because it was low effort and I felt like they integrated fairly cleanly.  I also thought that if any of these do happen to overflow or underflow (possibly in unit tests or something artificial), the error condition may be more obvious.

But I also was very much on the fence about the added indirection since I also did not think there is a reasonable security risk here.

I appreciate the feedback, I will go ahead and just close this, thank you!
